src/plots/plots.js

```diff
@@ -1144,9 +1144,13 @@ plots.supplyTraceDefaults = function(traceIn, traceOut, colorIndex, layout, trac
         coerce('ids');
 
         if(Registry.traceIs(traceOut, 'showLegend')) {
+            traceOut._dfltShowLegend = true;
             coerce('showlegend');
             coerce('legendgroup');
         }
+        else {
+            traceOut._dfltShowLegend = false;
```

Comment:
For the record: I added `_dfltShowLegend` because there's otherwise no robust way to tell from just `traceIn` and `traceOut` what the default is... so there would be some situations where we would NOT show a legend (eg scatter + colored contour) but then setting `showlegend: false` explicitly in the trace that already defaults to `showlegend: false` would make the legend appear 🤔

Reply:
yeah, that's what I expected. I don't think there's a nicer way to do this than what you coded in this commit :ok_hand: